bookkeeper-server/src/test/java/org/apache/bookkeeper/util/collections/ConcurrentLongLongPairHashMapTest.java

```diff
@@ -113,6 +113,22 @@ public void testRemove() {
         assertTrue(map.isEmpty());
     }
 
+    @Test
+    public void testExpandAndShrink() {
+        ConcurrentLongLongPairHashMap map = new ConcurrentLongLongPairHashMap(2, 1);
+        assertTrue(map.put(1, 1, 11, 11));
+        assertTrue(map.put(2, 2, 22, 22));
+        assertTrue(map.put(3, 3, 33, 33));
+
+        // expand hashmap
+        assertTrue(map.capacity() == 8);
+
+        assertTrue(map.remove(1, 1, 11, 11));
```

Comment:
Please assert that the map did not shrink here, otherwise we cannot state when it shrank

Reply:
OK

Reply:
Fixed.